src/transformationBuffer.cpp

```diff
@@ -69,131 +73,122 @@ bool TransformationBuffer::newestEntry(StampedTransformation &te) const {
     buffer.insert(++iterA, te);
     pruneStorage();
     return true;
 }
 
-void TransformationBuffer::pruneStorage(){
+void TransformationBuffer::pruneStorage() {
 
-    // NOTE: assertion just during development
-    // pruneStorage should never be called when the buffer is empty
+    // NOTE: Assertion just during development.
+    // Should never be called when the buffer is empty.
     assert(!buffer.empty());
 
     Timestamp mostRecent = ((StampedTransformation)buffer.back()).time;
     buffer.remove_if([&mostRecent, this](const StampedTransformation &te){return te.time + storageTime < mostRecent;});
 
-    // WARNING: TEST ME!
-    // to many entries, remove the oldest ones..
+    // To many entries, remove the oldest ones.
     if(buffer.size() > maxNumberOfEntries){
         while(buffer.size() > maxNumberOfEntries)
             buffer.pop_front();
     }
-
-
-    return;
-}
-
-void TransformationBuffer::printCurrentBufferToSTDOut() const{
-    for(StampedTransformation te : buffer )
-        cout << te;
 }
 
 bool TransformationBuffer::entryAt(StampedTransformation &te) const {
 
-    // return false if buffer is empty
+    // Return false if buffer is empty.
     if(buffer.empty())
         return false;
 
-    // return the newest entry if queried for en even newer one
+    // Return the newest entry if queried for en even newer one.
     if(((StampedTransformation)buffer.back()).time <= te.time){
         te = buffer.back();
         return true;
     }
 
-    // return the oldest entry queried for an even older one
+    // Return the oldest entry queried for an even older one.
     if(te.time <= ((StampedTransformation)buffer.front()).time){
         te = buffer.front();
         return true;
     }
 
-    // search the two closest entries
+    // Search the two closest entries.
     auto iterA = adjacent_find(buffer.begin(), buffer.end(), [&te](const StampedTransformation &el, const StampedTransformation &er){ return (el.time < te.time && te.time < er.time); });
     StampedTransformation tl = *iterA; StampedTransformation tr = *(++iterA);
 
-    // return left entry if distance between entries is to small
+    // Return left entry if distance between entries is to small.
     if(tr.time - tl.time < minDistForInterpolation){
         te = tl;
         return true;
     }
 
-    // interpolate
     interpolate(tl, tr, te);
 
     return true;
 }
 
-void TransformationBuffer::interpolate(const StampedTransformation &el, const StampedTransformation &er, StampedTransformation &res) const{
+void TransformationBuffer::interpolate(const StampedTransformation &el, const StampedTransformation &er, StampedTransformation &res) const {
 
     milliseconds abs = duration_cast<milliseconds>(er.time-el.time);
     milliseconds lt = duration_cast<milliseconds>(res.time-el.time);
 
-    float ratio = (float) lt.count()/abs.count();
+    double ratio = (double) lt.count()/abs.count();
```

Comment:
No, since the function interpolate just is called if the timestamps of the left and the right entry are at least a minimal distance apart.